spec/javascripts/unit/admin/enterprises/directives/terms_and_conditions_warning_spec.js.coffee

```diff
@@ -0,0 +1,18 @@
+describe "termsAndConditionsWarning", ->
+  element = null
+  templatecache = null
+
+  beforeEach ->
+    module('admin.enterprises')
+
+    inject ($rootScope, $compile, $templateCache) ->
+      templatecache = $templateCache
+      el = angular.element("<input terms-and-conditions-warning=\"true\"></input>")
+      element = $compile(el)($rootScope)
+      $rootScope.$digest()
+
+  describe "terms and conditions warning", ->
+    it "should load template", ->
+      spyOn(templatecache, 'get')
+      element.triggerHandler('click');
+      expect(templatecache.get).toHaveBeenCalledWith('admin/modals/terms_and_conditions_warning.html')
```

Comment:
here I made some progress with testing directives. Here we trigger an event and see that the template was loaded :+1: 
BUT testing that the warning is actually shown to the user when user click on "Chose file" is impossible to test, either in this directive test or in the feature test, because the file input is a browser protected feature.